asyncpg/protocol/prepared_stmt.pyx

```diff
@@ -84,12 +87,22 @@ cdef class PreparedStatementState:
                 if codec is None or not codec.has_decoder():
                     missing.add(rdesc[3])
 
+        if disable_custom_codec:
+            self.settings.use_custom_codec(True)
+
         return missing
 
-    cpdef _init_codecs(self):
+    cpdef _init_codecs(self, bint disable_custom_codec):
+
+        if disable_custom_codec:
+            self.settings.use_custom_codec(False)
```

Comment:
No need to affect global state here, just pass the argument down to `_ensure_...` and down to `get_codec`.

Reply:
Yeah, that makes sense - I'm only feeling a bit panic adding so many passing-downs.